rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -12,24 +12,80 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <rcpputils/filesystem_helper.hpp>
+#include <rcpputils/get_env.hpp>
 #include <rcutils/allocator.h>
+#include <rcutils/env.h>
 #include <rcutils/error_handling.h>
 #include <rcutils/logging.h>
-#include "rcl_logging_spdlog/logging_interface.h"
+
+#include <fstream>
+#include <string>
+
+#include "fixtures.hpp"
 #include "gtest/gtest.h"
+#include "rcl_logging_spdlog/logging_interface.h"
 
-static void * bad_malloc(size_t, void *)
+namespace fs = rcpputils::fs;
+
+const int logger_levels[] =
 {
-  return nullptr;
-}
+  RCUTILS_LOG_SEVERITY_UNSET,
+  RCUTILS_LOG_SEVERITY_DEBUG,
+  RCUTILS_LOG_SEVERITY_INFO,
+  RCUTILS_LOG_SEVERITY_WARN,
+  RCUTILS_LOG_SEVERITY_ERROR,
+  RCUTILS_LOG_SEVERITY_FATAL,
+};
+
+// This is a helper class that resets an environment
+// variable when leaving scope
+class RestoreEnvVar
+{
+public:
+  explicit RestoreEnvVar(const std::string & name)
+  : name_(name),
+    value_(rcpputils::get_env_var(name.c_str()))
+  {
+  }
+
+  ~RestoreEnvVar()
+  {
+    if (!rcutils_set_env(name_.c_str(), value_.c_str())) {
+      std::cerr << "Failed to restore value of environment variable: " << name_ << std::endl;
+    }
+  }
 
-TEST(logging_interface, init_invalid)
+private:
+  const std::string name_;
+  const std::string value_;
+};
+
+// TODO(cottsay): Remove when ros2/rcpputils#63 is resolved
+static fs::path current_path()
 {
-  rcutils_allocator_t allocator = rcutils_get_default_allocator();
-  rcutils_allocator_t bad_allocator = rcutils_get_default_allocator();
-  rcutils_allocator_t invalid_allocator = rcutils_get_zero_initialized_allocator();
-  bad_allocator.allocate = bad_malloc;
+  char * cwd;
+#ifdef _WIN32
+#ifdef UNICODE
+#error "rcpputils::fs does not support Unicode paths"
+#endif
+  cwd = _getcwd(NULL, 0);
+#else
+  cwd = getcwd(NULL, 0);
+#endif
+  if (nullptr == cwd) {
+    std::error_code ec{errno, std::system_category()};
+    errno = 0;
```

Comment:
@cottsay I'd rather not change `errno`'s value.

Reply:
I'm going to leave this as-is to align with the behavior that will eventually be in `rcpputils`. See https://github.com/ros2/rcpputils/pull/63#discussion_r435543631 for details.